### PR TITLE
Update validate regex to exclude hex strings, integers and floats

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207792893525794

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -506,7 +506,11 @@ function validate(...args: string[]): void {
     if (arg.split('-').length - 1 > 1) {
       throw new Error(`Invalid number: more than one negative sign '${arg}'`)
     }
-    if (/[^\d.\-x]/.test(arg)) {
+    if (
+      /^(?!\s*-?(0x[0-9A-Fa-f]+|(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?\s*)$).+/.test(
```
Comment (samholmes) [inline src/index.ts:510]: Use `!/^-?(?:0x[0-9A-Fa-f]+|(?:\d+(?:\.\d*)?|\.\d+))$/.text(arg)` which removes scientific notation, doesn't use a negative look ahead, and removes the white space checks.

File: src/index.ts [hunk 2]
```diff
@@ -506,7 +506,11 @@ function validate(...args: string[]): void {
     if (arg.split('-').length - 1 > 1) {
```
Comment (samholmes) [inline src/index.ts:506]: Is two negative signs possible given scientific notation like `-12e-2`? I doubt it, but if we ever see this error message crop up, we know possibly why.

**Review by samholmes — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -213,6 +213,9 @@ export const round = (x1: string | number, precision: number): string =>
 
 export function toBns(n: number | string): string {
   let out = typeof n === 'number' ? n.toString() : n.replace(/^\s+|\s+$/g, '')
+  if (out === '') {
+    out = '0'
+  }
```
Comment (samholmes) [inline src/index.ts:218]: This should be an error case rather than coercing white-space character strings to a big number string

**Review by samholmes — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (peachbits) [reply src/index.ts:218]: biggystring already handles empty string as zero and there are tests for it, for better or worse.

Comment (samholmes) [reply src/index.ts:218]: I guess this has been to spec for a while. 🤷‍♂️

**Review by samholmes — APPROVED** [review]

**Pull request merged**